### PR TITLE
modified spark example to use kubectl exec to interact with cluster a…

### PR DESCRIPTION
**Pull request by gurvindersingh**
…nd create spark driver pod

You can change the repo in the driver spec file to gcr.io, once the pull request is accepted and docker image pushed to official repository.

### CODE REVIEW

**Comment by gurvindersingh** [comment]
I signed it!

**Comment by brendandburns** [comment]
@gurvindersingh 
I don't see you in our CLA records, what email did you use?

We also moved a bunch of docs in the lead up to 1.0, you may need to recreate this PR.  Sorry!

**Comment by gurvindersingh** [comment]
@brendandburns I have used this email address to sign the CLA <gurvindersinghdahiya@gmail.com> let me know if you can find, otherwise I can give it another try to sign. I will create/rebase the PR to the current master.

**Comment by brendandburns** [comment]
CLA verified.  I think you need to associate your email with your github (or add your github to clasigners) in order for this to be automated.

Needs a rebase.

File: examples/spark/spark-driver.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (brendandburns) [inline examples/spark/spark-driver.json]: any chance you can convert this to YAML instead of JSON?

File: examples/spark/images/driver/start.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+echo "$SPARK_MASTER_SERVICE_HOST spark-master" >> /etc/hosts
+echo "SPARK_LOCAL_HOSTNAME=$(hostname -i)" >> /opt/spark/conf/spark-env.sh
+echo "MASTER=spark://spark-master:$SPARK_MASTER_SERVICE_PORT" >> /opt/spark/conf/spark-env.sh
+
+while true; do
+sleep 100
```
Comment (brendandburns) [inline examples/spark/images/driver/start.sh]: indent:
```sh
while true; do
  sleep 100
done
```

File: examples/spark/spark-driver.json (hunk 1 above)
Comment (gurvindersingh) [inline examples/spark/spark-driver.json]: All other specs in spark example are also in JSON, so should I convert all to YAML or leave as it is. I see there are few examples which have specs in YAML and whereas others in JSON. So any policy towards specs format ?

**Comment by gurvindersingh** [comment]
Rebased and addressed comment about indent, regarding JSON -> YAML.. will wait for @brendandburns reply.

File: examples/spark/spark-driver.json (hunk 1 above)
Comment (brendandburns) [inline examples/spark/spark-driver.json]: Preference for YAML.  If you'd rather defer to a different PR, that's fine.

File: examples/spark/images/driver/start.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (brendandburns) [inline examples/spark/images/driver/start.sh]: you need to add standard copyright boilerplate here.

**Comment by brendandburns** [comment]
Need copyright boilerplate on `start.sh` otherwise, LGTM.

File: examples/spark/spark-driver.json (hunk 1 above)
Comment (gurvindersingh) [inline examples/spark/spark-driver.json]: I will send another pull request which will converting all the JSON spec files in spark example to YAML

**Comment by gurvindersingh** [comment]
added copyright boilerplate on ```start.sh```

**Comment by brendandburns** [comment]
You need to run `./hack/run-gendocs.sh`

(sorry for the complexity!)

**Comment by gurvindersingh** [comment]
i ran ```./hack/run-gendocs.sh```and committed the changes to pull request. let me know if anything more is needed.

**Comment by brendandburns** [comment]
LGTM, thanks!

**Comment by ixdy** [comment]
This test is failing continuously on GCE Jenkins.

```
INFO: Running '/jenkins-master-data/jobs/kubernetes-e2e-gce-examples/workspace/kubernetes/platforms/linux/amd64/kubectl --server=https://130.211.183.151 --kubeconfig=/var/lib/jenkins/jobs/kubernetes-e2e-gce-examples/workspace/.kube/config log spark-driver spark-driver --namespace=e2e-tests-examples-ou9nm'
INFO: 
[AfterEach] Examples e2e
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/examples.go:69
STEP: Destroying namespace for this suite e2e-tests-examples-ou9nm

• Failure [538.277 seconds]
Examples e2e
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/examples.go:528
  [Skipped][Example]Spark
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/examples.go:203
    should start spark master, driver and workers [It]
    /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/examples.go:202

    Expected error:
        <*errors.errorString | 0xc2082656a0>: {
            s: "Failed to find \"Starting Spark driver at\", last result: \"\"",
        }
        Failed to find "Starting Spark driver at", last result: ""
    not to have occurred

    /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/examples.go:193
```

Full build log/artifacts from a recent run are available here: https://cloud.google.com/console/storage/kubernetes-jenkins/logs/kubernetes-e2e-gce-examples/465/

**Pull request merged**